### ISSUE TITLE
_parse_help missing short options when long option is present

### ISSUE THREAD

**Issue by vt-alt** [scop/bash-completion#831]
## Describe the bug

`_parse_help` function does not show all options missing short options when they are show before long option.

## To reproduce

`eu-strip --help` have lines like this:
```
  -o, --output=FILE          Place stripped output into FILE
...
  -p, --preserve-dates       Copy modified/access timestamps to the output
```

```
$ git clone https://github.com/scop/bash-completion
$ cd bash-completion/
bash-completion (master)$ . ./bash_completion
bash-completion (master)$ _parse_help eu-strip
-f
-F
--output=
--strip-debug
--keep-section=
--permissive
--preserve-dates
--reloc-debug-sections
--reloc-debug-sections-only
--remove-comment
--remove-comment
--remove-section=
--strip-sections
--help
--usage
--version
```
Standalone short options are shown.

## Expected behavior

There is `--output=` but no `-o`. It would be good to see `-o` (and other similar options) there too.

## Versions (please complete the following information)

- [x] Operating system name/distribution and version: ALT Linux
- [x] bash version, `echo "$BASH_VERSION"`: 4.4.23(1)-release
- [x] bash-completion version, `(IFS=.; echo "${BASH_COMPLETION_VERSINFO[*]}")`: 2.11

## Additional context

By the comment in the code:
```
bash_completion:1177:            # transform "-f FOO, --foo=FOO" to "-f , --foo=FOO" etc
```
I guess it tries to handle this, but misses somehow.

**Comment by vt-alt** [comment]
IC. Thanks for the answer!
ps. Well, sometimes short options are preferable because users accustomed to them.

**Comment by scop** [comment]
Yes. Users who are accustomed to and prefer them can just type them -- <kbd>-X</kbd> takes fewer keystrokes than <kbd>-Tab</kbd> + _then_ typing the desired one. And I don't think us listing the short ones would be generally useful, beacuse as noted, the users are already accustomed to them and know what they're doing anyway and can just type the option, whereas users who don't might find it harder to find what they're looking for if we offered both short and long ones. Note that we do try to support arguments to short options too.
